src/core/backbone_router/local.hpp

```diff
@@ -162,9 +163,49 @@ class Local : public InstanceLocator
      */
     void UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRouterConfig &aConfig);
 
+    /**
+     * This method gets the Domain Prefix configuration.
+     *
+     * @param[out]  aConfig  A reference to the Domain Prefix configuration.
+     *
+     * @retval OT_ERROR_NONE       Successfully got the Domain Prefix configuration.
+     * @retval OT_ERROR_NOT_FOUND  No Domain Prefix was configured.
+     *
+     */
+    otError GetDomainPrefix(NetworkData::OnMeshPrefixConfig &aConfig);
+
+    /**
+     * This method removes the local Domain Prefix configuration.
+     *
+     * @param[in]  aPrefix A reference to the IPv6 Domain Prefix.
+     *
+     * @retval OT_ERROR_NONE          Successfully removed the Domain Prefix.
+     * @retval OT_ERROR_INVALID_ARGS  @p aPrefix is invalid.
+     * @retval OT_ERROR_NOT_FOUND     No Domain Prefix was configured or @p aPrefix doesn't match.
+     *
+     */
+    otError RemoveDomainPrefix(const otIp6Prefix &aPrefix);
+
+    /**
+     * This method sets the local Domain Prefix configuration.
+     *
+     * @param[in]  aConfig A reference to the Domain Prefix configuration.
+     *
+     */
+    void SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig);
+
 private:
     void    SetState(BackboneRouterState aState);
     otError RemoveService(void);
+    void    AddDomainPrefixToNetworkData(void);
+    void    RemoveDomainPrefixFromNetworkData(void);
```

Comment:
suggestion: How about since these are method in `NetworkData::Local` class, we omit the `ToNetworkData`, `FromNetworkData` in the method names and use `AddDomainPrefix`/`RemoveDomainPrefix`?

Reply:
I would prefer the current names which are more descriptive and accurate. Thoughts @librasungirl

Reply:
@abtink, this is the `BackboneRouter::Local` class.

Reply:
Let's keep the existing names for readability as clarified by @simonlingoogle ? @abtink

Reply:
SGTM. Let's keep the name. I mistook this file as `NetworkData::Local`... :)